tests/Silverback.Integration.Tests.E2E/Kafka/CallbacksTests.cs

```diff
@@ -151,6 +156,177 @@ public async Task EndpointsConfiguredCallback_MessagePublished()
+                                                config.GroupId = "consumer1";
+                                                config.EnablePartitionEof = true;
+                                            })))
+                        .AddIntegrationSpyAndSubscriber())
+                .Run();
+
+            var kafkaTestingHelper = Host.ServiceProvider.GetRequiredService<IKafkaTestingHelper>();
+
+            var publisher = Host.ScopedServiceProvider.GetRequiredService<IEventPublisher>();
+            var callbackHandlers = Host.ScopedServiceProvider
+                .GetServices<IBrokerCallback>()
+                .Where(service => service is KafkaEndOfTopicPartitionReachedCallback)
+                .ToList();
+
+            callbackHandlers.Should().HaveCount(1);
+            callbackHandlers.Cast<KafkaEndOfTopicPartitionReachedCallback>().Should().OnlyContain(handler => handler.EndOfPartitionReachedCount == 2);
+
+            await kafkaTestingHelper.Broker.Consumers[0].DisconnectAsync();
+
+            await publisher.PublishAsync(messageEventOne);
+            await publisher.PublishAsync(messageEventTwo);
+
+            await kafkaTestingHelper.Broker.Consumers[0].ConnectAsync();
+
+            await kafkaTestingHelper.WaitUntilAllMessagesAreConsumedAsync();
+
+            kafkaTestingHelper.Spy.InboundEnvelopes.Should().HaveCount(2);
+            callbackHandlers.Cast<KafkaEndOfTopicPartitionReachedCallback>().Should().OnlyContain(handler => handler.EndOfPartitionReachedCount == 4);
+
+            await kafkaTestingHelper.Broker.Consumers[0].DisconnectAsync();
+
+            for (var i = 0; i < 2; i++)
+            {
+                await publisher.PublishAsync(messageEventOne);
+                await publisher.PublishAsync(messageEventTwo);
+            }
+
+            await kafkaTestingHelper.Broker.Consumers[0].ConnectAsync();
+
+            await kafkaTestingHelper.WaitUntilAllMessagesAreConsumedAsync();
+
+            kafkaTestingHelper.Spy.InboundEnvelopes.Should().HaveCount(6);
+            callbackHandlers.Cast<KafkaEndOfTopicPartitionReachedCallback>().Should().OnlyContain(handler => handler.EndOfPartitionReachedCount == 6);
+
+            await kafkaTestingHelper.Broker.Consumers[0].DisconnectAsync();
+
+            for (var i = 0; i < 3; i++)
+            {
+                await publisher.PublishAsync(messageEventOne);
+                await publisher.PublishAsync(messageEventTwo);
+            }
+
+            await kafkaTestingHelper.Broker.Consumers[0].ConnectAsync();
+
+            await kafkaTestingHelper.WaitUntilAllMessagesAreConsumedAsync();
+
+            kafkaTestingHelper.Spy.InboundEnvelopes.Should().HaveCount(12);
+            callbackHandlers.Cast<KafkaEndOfTopicPartitionReachedCallback>().Should().OnlyContain(handler => handler.EndOfPartitionReachedCount == 8);
+
+            await kafkaTestingHelper.Broker.Consumers[0].DisconnectAsync();
+
+            for (var i = 0; i < 5; i++)
```

Comment:
I understand the reason to disconnect -> publish -> disconnect again -> publish again. This is correct to ensure that the EOF message gets published every time. But why do it another 2 extra times and with an incremental number of messages? What do you want to ensure?